### ISSUE TITLE
keyboardHandler.injectRawKeyboardInput stops working in NVDA settings menu

### ISSUE THREAD

**Issue by tibabit** [nvaccess/nvda#8020]
I have noticed the `keyboardHandler.injectRawKeyboardInput`  stops working once I open NVDA menu by pressing insert+n. How ever I am able to navigate through it using the keyboard on my laptop, and once I exit NVDA menu by pressing escape on the keyboard in my laptop `keyboardHandler.injectRawKeyboardInput` starts working again. 

Any one knows if this is known issue? or anyone facing the similar issue?

### Steps to reproduce:
- start NVDA
- press insert+n on keyboard of your laptop
- press up or down arrow keys on laptop (it's  working)
- press up or down arrow keys through `keyboardHandler.injectRawKeyboardInput` (it's now working)

### Expected behavior:
user should be able to navigate up and down in NVDA menu

### Actual behavior:
Nothing happens

### System configuration:
NVDA version:
2017.1.0dev and the latest version 2017.4

NVDA Installed or portable: 
**2017.1.0dev** running from source code
**2017.4** installed version

Windows version:
Windows 7 and 10

Name and version of other software in use when reproducing the issue:

### Other questions:

Does the issue still occur after restarting your PC? It happens always

Have you tried any other versions of NVDA?
No

**Comment by leonardder** [comment]
[**@tibabit**](https://github.com/tibabit) commented on [21 Feb 2018, 15:38 CET](https://github.com/nvaccess/nvda/issues/8020#issue-298994638 "2018-02-21T14:38:38Z - Replied by Github Reply Comments"):
> - press up or down arrow keys through `keyboardHandler.injectRawKeyboardInput` (it's now working)

I assume you meant "not working" here? Could you also elaborate on how you use `injectRawKeyboardInput`?

**Comment by tibabit** [comment]
Yes, I meant "not working here".

I am using following code snippet to inject keys-

	def injectKey(cls, vk, isPress, extended):
		scancode = MapVirtualKey(vk, 0)
		keyboardHandler.injectRawKeyboardInput(isPress, scancode, extended)

I am getting virtual key codes of the keys from the [this](https://msdn.microsoft.com/en-us/library/windows/desktop/dd375731(v=vs.85).aspx)
and for extended flag I am referring to the [this](http://www.computer-engineering.org/ps2keyboard/scancodes1.html)

Note that. This snippet works fine at all places I have test, except this one.

**Comment by leonardder** [comment]
Are you sure you don't have "Handle keys from other applications" 
disabled in a particular config profile?

**Comment by tibabit** [comment]
Yes it's not disabled. Like i said it's working at all other places, I tested.

**Comment by leonardder** [comment]
I'm sorry for the time it took to come back to this. I think I have an idea what causes this.

@tibabit: What happens if you:

1. create a gesture using keyboardHandler.KeyboardInputGesture.fromName
2. Execute gesture.send()

**Comment by Adriani90** [comment]
@tibabit do you still encounter this issue in NVDA 2019.1.1? Have you already tried to consider the suggestion form Leonard above?

**Comment by tibabit** [comment]
I have not been able to work on it for quite some time.

**Comment by Adriani90** [comment]
Given the missing updates and lack of updated test results, I am closing this issue as abandoned. Feel free to comment on it if you are still having it and we can reopen.
